Source/pfile.h

```diff
@@ -8,12 +8,76 @@
 #include "DiabloUI/diabloui.h"
 #include "player.h"
 
+#ifdef UNPACKED_SAVES
+#include "utils/file_util.h"
+#else
+#include "mpq/mpq_reader.hpp"
+#include "mpq/mpq_writer.hpp"
+#endif
+
 namespace devilution {
 
 #define MAX_CHARACTERS 99
 
 extern bool gbValidSaveFile;
 
+#ifdef UNPACKED_SAVES
+struct SaveReader {
+	explicit SaveReader(std::string &&dir)
+	    : dir_(std::move(dir))
+	{
+	}
+
+	const std::string &dir() const
+	{
+		return dir_;
+	}
+
+	std::unique_ptr<byte[]> ReadFile(const char *filename, std::size_t &fileSize, int32_t &error);
+
+	bool HasFile(const char *path)
+	{
+		return ::devilution::FileExists((dir_ + path).c_str());
+	}
+
+private:
+	std::string dir_;
+};
+
+struct SaveWriter {
+	explicit SaveWriter(std::string &&dir)
+	    : dir_(std::move(dir))
+	{
+	}
+
+	bool WriteFile(const char *filename, const byte *data, size_t size);
+
+	bool HasFile(const char *path)
+	{
+		return ::devilution::FileExists((dir_ + path).c_str());
+	}
+
+	void RenameFile(const char *from, const char *to)
+	{
+		::devilution::RenameFile((dir_ + from).c_str(), (dir_ + to).c_str());
+	}
+
+	void RemoveHashEntry(const char *path)
+	{
+		RemoveFile((dir_ + path).c_str());
+	}
+
+	void RemoveHashEntries(bool (*fnGetName)(uint8_t, char *));

```

Comment:
Ideally the method would be called RemoveFile(s) and for the MpqWriter it does it by removing hash entires.